src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerChangeService.java

```diff
@@ -150,17 +150,15 @@ private BeneficialOwnerChange<CorporateBeneficialOwnerPsc> covertBeneficialOwner
     Pair<PscApi, PrivateBoDataApi> publicPrivateBoPair = publicPrivateBo.get(
         beneficialOwnerCorporateDto.getChipsReference());
 
-    if (publicPrivateBoPair == null) {
+    if (publicPrivateBoPair == null || publicPrivateBoPair.getRight() == null) {
```

Comment:
If not having private data is meant to be an issue it might be worth changing the log message.

Currently, this will log that the entire pair wasn't found making anyone looking at it assume that there wasn't anything in the HashMap

Reply:
Or obviously making it a separate if statement with a different log message